test/unit/org/apache/cassandra/metrics/DecayingEstimatedHistogramReservoirTest.java

```diff
@@ -604,6 +616,110 @@ public void testSize()
+            AtomicBoolean stop = new AtomicBoolean(false);
+            AtomicBoolean failed = new AtomicBoolean(false);
+            TestClock clock = new TestClock();
+
+            DecayingEstimatedHistogramReservoir histogram = new DecayingEstimatedHistogramReservoir(clock);
+            ExecutorService executors = Executors.newFixedThreadPool(2 + UPDATE_THREADS);
+
+            for (int i = 0; i < UPDATE_THREADS; i++)
+            {
+                executors.submit(() -> {
+                    while (!stop.get() && !Thread.currentThread().isInterrupted())
+                    {
+                        // a mischievous usage pattern to quickly trigger the
+                        // CASSANDRA-19365 race condition;
+                        // the load has 95% of 42, and only 5% of the time it's 1109
+                        // and yet, the histogram may be convinced for a long time that
+                        // the p99 is 42 or that the p50 is 1109
+                        for (int sampleIdx = 0; sampleIdx < 900; sampleIdx++)
+                            histogram.update(42);
+
+                        for (int sampleIdx = 0; sampleIdx < 50; sampleIdx++)
+                        {
+                            // add some noise so that low value samples do not race with the same likelyhood as the high value samples
+                            Uninterruptibles.sleepUninterruptibly(ThreadLocalRandom.current().nextInt(1, 10), MILLISECONDS);
+                            histogram.update(1109);
+                        }
+                    }
+                });
+            }
+            // clock update thread
+            executors.submit(() -> {
+                while (!stop.get() && !Thread.currentThread().isInterrupted())
+                {
+                    Uninterruptibles.sleepUninterruptibly(1, TimeUnit.MILLISECONDS);
+                    // x1000 speedup so that we hit rescale interval every 30 minutes / 1000 = 1.8s
+                    clock.addMillis(1000);
+                }
+            });
+            // percentiles check thread
+            executors.submit(() -> {
+                // how many times in a row p99 was suspiciously low or P50 suspiciously high
+                int consecutiveInvalidPercentiles = 0;
+
+                // how often to check the percentiles
+                int iterationDelayMillis = 100;
+
+                for (int i = 0; i < maxTestDurationMillis / iterationDelayMillis; i++)
+                {
+                    Uninterruptibles.sleepUninterruptibly(iterationDelayMillis, MILLISECONDS);
+                    Snapshot snapshot = toSnapshot.apply(histogram);
+                    double p99 = snapshot.getValue(0.99);
+                    double p50 = snapshot.getValue(0.50);
+                    logger.info("{} p99: {}", clock.now() / 1_000_000, p99);
+                    logger.info("{} p50: {}", clock.now() / 1_000_000, p50);
```

Comment:
nit: Could we just replace this with a single line?

```
logger.info("{}s, p50: {}, p99: {}", TimeUnit.NANOSECONDS.toSeconds(clock.now()), p50, p99);
```

Reply:
Fixed.